src/services/MetricsService/tests/MetricsFilter.spec.ts

```diff
@@ -0,0 +1,190 @@
+    expect(filteredEvent2).not.toBeNull();
+
+    const filteredEvent3 = await sut.getDelayedOnboardedEventIfPublishable('n/a', 'n/a');
+    expect(filteredEvent3).toBeNull();
+  });
+
+  it('otk-no-date event is only published if in exposed state', async () => {
+    const filteredEvent1 = await sut.filterEvent({type: EventTypeMetric.OtkNoDate, exposureHistory: []});
+    expect(filteredEvent1).toBeNull();
+
+    const filteredEvent2 = await sut.filterEvent({type: EventTypeMetric.OtkNoDate, exposureHistory: [1]});
+    expect(filteredEvent2).not.toBeNull();
+  });
+
+  it('en-toggle event (set to enabled) is only published if user is 24 hours after onboarding date', async () => {
+    today = new OriginalDate('2019-01-02T23:00:00.000Z');
+
+    const filteredEvent1 = await sut.filterEvent({
+      type: EventTypeMetric.EnToggle,
+      state: true,
+      onboardedDate: new OriginalDate('2019-01-02T12:00:00.000Z'),
+    });
+    expect(filteredEvent1).toBeNull();
+
+    today = new OriginalDate('2019-01-03T13:00:00.000Z');
+
+    const filteredEvent2 = await sut.filterEvent({
+      type: EventTypeMetric.EnToggle,
+      state: true,
+      onboardedDate: new OriginalDate('2019-01-02T12:00:00.000Z'),
+    });
+    expect(filteredEvent2).not.toBeNull();
+  });
+
+  it('exposed-clear event is only published if in exposed state', async () => {
+    const filteredEvent1 = await sut.filterEvent({
+      type: EventTypeMetric.ExposedClear,
+      exposureStatus: {
+        type: ExposureStatusType.Monitoring,
+        lastChecked: null,
+        ignoredSummaries: null,
+      },
+    });
+    expect(filteredEvent1).toBeNull();
+
+    const filteredEvent2 = await sut.filterEvent({
+      type: EventTypeMetric.ExposedClear,
+      exposureStatus: {
+        type: ExposureStatusType.Exposed,
+        summary: {
+          attenuationDurations: [],
+          daysSinceLastExposure: 1,
+          lastExposureTimestamp: 1,
+          matchedKeyCount: 1,
+          maximumRiskScore: 1,
+        },
+        exposureDetectedAt: 1,
+        notificationSent: null,
+        lastChecked: null,
+        ignoredSummaries: null,
+      },
+    });
+    expect(filteredEvent2).not.toBeNull();
+  });
+
+  it('active-user event is only published once a day', async () => {
+    today = new OriginalDate('2019-01-01T12:00:00.000Z');
+
+    const filteredEvent1 = await sut.filterEvent({type: EventTypeMetric.ActiveUser});
+    expect(filteredEvent1).not.toBeNull();
+
+    const filteredEvent2 = await sut.filterEvent({type: EventTypeMetric.ActiveUser});
+    expect(filteredEvent2).toBeNull();
+
+    today = new OriginalDate('2019-01-02T00:00:00.000Z');
+
+    const filteredEvent3 = await sut.filterEvent({type: EventTypeMetric.ActiveUser});
+    expect(filteredEvent3).not.toBeNull();
+  });
+
+  it('background-check event is only published once a day', async () => {
+    today = new OriginalDate('2019-01-01T12:00:00.000Z');
```

Comment:
Should make it more obvious what dates we're on / moving to 

Maybe day1, day2, day 3 or something to make it easier to see in the tests what days we're expecting to see in the logs etc... just something to make it more obvious.

Reply:
`today` is the global variable name used to dynamically change the mock of current date. I guess I can add some comments to guide people through the test case.